function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import ExtendableError from './ExtendableError';

var AlreadyInUseError = function (_ExtendableError) {
  _inherits(AlreadyInUseError, _ExtendableError);

  function AlreadyInUseError() {
    _classCallCheck(this, AlreadyInUseError);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    var _this = _possibleConstructorReturn(this, (AlreadyInUseError.__proto__ || Object.getPrototypeOf(AlreadyInUseError)).call(this, args));

    var i = 0;
    var entityName = null;
    var constraints = [];

    args.forEach(function (arg) {
      if (i === 0 && typeof arg === 'string') {
        entityName = arg;
      } else if (typeof arg === 'string') {
        constraints.push(arg);
      }

      i++;
    });

    if (args[0] !== undefined && typeof args[0] === 'string') {
      _this.message = constraints.length > 0 ? 'The specified \'' + entityName + '\' value is already in use for: \'' + constraints.join(', ') + '\'' : 'The specified \'' + entityName + '\' value is already in use';
    }

    return _this;
  }

  return AlreadyInUseError;
}(ExtendableError);

export default AlreadyInUseError;